File: Lighthouse.Frontend/src/models/AppSettings/RefreshSettings.ts
import { z } from "zod";

export interface IRefreshSettings {
	interval: number;
	refreshAfter: number;
	startDelay: number;
}

export const RefreshSettingsSchema = z.object({
	interval: z.number(),
	refreshAfter: z.number(),
	startDelay: z.number(),
});

export class RefreshSettings implements IRefreshSettings {
	interval: number;
	refreshAfter: number;
	startDelay: number;

	constructor(interval: number, refreshAfter: number, startDelay: number) {
		this.interval = interval;
		this.refreshAfter = refreshAfter;
		this.startDelay = startDelay;
	}
}

File: Lighthouse.Backend/Lighthouse.Backend/Models/AppSettings/RefreshSettings.cs
using System.Text.Json.Serialization;

namespace Lighthouse.Backend.Models.AppSettings
{
    public class RefreshSettings
    {
        [JsonRequired]
        public int Interval { get; set; }

        [JsonRequired]
        public int RefreshAfter { get; set; }

        [JsonRequired]
        public int StartDelay { get; set; } 
    }
}
